### PR TITLE
Don't change colors when TERM=dumb

### PR DESCRIPTION
**Pull request by raylai**
Please play nicer with dumb terminals, thank you!

### CODE REVIEW

**Comment by daviddengcn** [comment]
Thanks for the commit.

I'm not familiar with the "TERM=dumb" convention can you refer me any link that helps?

**Comment by raylai** [comment]
On Tue, 26 Apr 2016 22:22:36 -0700
David Deng <notifications@github.com> wrote:

> Thanks for the commit.
> 
> I'm not familiar with the "TERM=dumb" convention can you refer me any link that helps?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/daviddengcn/go-colortext/pull/5#issuecomment-214972607

https://en.wikipedia.org/wiki/Computer_terminal#Dumb_terminals
https://emacs.stackexchange.com/questions/10137/what-does-term-dumb-in-eshell-mean
https://github.com/facebook/flow/pull/263

And specifically I use plan9port's Acme, which sets TERM=dumb (as it just displays escape sequences as raw garbage).

File: ct_ansi.go [hunk 1]
```diff
@@ -4,14 +4,21 @@ package ct
...
 
 func resetColor() {
+	if os.Getenv("TERM") == "dumb" {
+		return
+	}
 	fmt.Print("\x1b[0m")
 }
 
 func ansiText(fg Color, fgBright bool, bg Color, bgBright bool) string {
+	if os.Getenv("TERM") == "dumb" {
```
Comment (daviddengcn) [inline ct_ansi.go]: No need to check here.

Comment (raylai) [inline ct_ansi.go]: Agreed. Also probably could remove the following if block as well:
```go
	if fg == None && bg == None {
		return ""
	}
```

Comment (daviddengcn) [inline ct_ansi.go]: This one is ok. It does not break the assumption ansiText is a pure function of its parameters.

**Comment by daviddengcn** [comment]
Thanks for the contribution!

File: ct_ansi.go [hunk 2]
```diff
@@ -4,10 +4,14 @@ package ct
 
 import (
 	"fmt"
+	"os"
 	"strconv"
 )
 
 func resetColor() {
+	if os.Getenv("TERM") == "dumb" {
```
Comment (shurcooL) [inline ct_ansi.go]: Question, do you expect the value of `os.Getenv("TERM")` to change during the execution of a process?

If not, consider factoring it out into a package level variable, e.g.:

```Go
var dumbTerminal = os.Getenv("TERM") == "dumb"
```

Then use it like this:

```Go
if dumbTerminal {
    return
}
```

Comment (daviddengcn) [inline ct_ansi.go]: No. I'm expecting those kind of control done by the user of the package itself.

**Pull request merged**